### DISCUSSION TITLE
Modelling a Mechanical One Way Component: A Ratchet

### DISCUSSION THREAD

**Discussion by OwenDavid** [Hopsan/hopsan#2041] [Ideas]
The hydraulic library contains a check valve which prevents flow in one direction.

An equivalent mechanical device is a ratchet which allow rotation in one direction and prevents it in the other. The wiki images shows a typical ratchet. This device is used in mechanical winches to allow a rope to be tensioned. When the winch motor stops the ratchet prevents the winch turning in the opposite direction and retains the tension on the rope.

![image](https://user-images.githubusercontent.com/66214518/103996207-cca2ce80-5199-11eb-9801-1de28b05dea2.png)

How could the TLM technique be used to model both a linear and rotational mechanical  one-way device? Would the equivalent code from the hydraulic check valve be relevant?

```


if (c1 > c2)
            {
            allow force/torque to transmit through component
            }
            else
            {
             do not allow force/torque to transmit through component
            }
```

Would the translational lossless connector be a good base component to adapt to achieve this?

**Comment by robbr48** [comment]
The equations should be sufficient, assuming the backlash does not need to be taken into account. It could also be possible to introduce backlash by only applying the limitation of velocity at certain angles, but I guess that would make it over-complicated for most users. It would require a "number of teeth" parameter.

It should work to base it on the lossless connector I think, as long as the velocity limitation is applied before integrating the position. Equations for a rotational ratchet should be the same as for the translational one.
